CorsixTH/Lua/entities/humanoids/staff/handyman.lua

```diff
@@ -0,0 +1,190 @@
+--[[ Copyright (c) 2009 Peter "Corsix" Cawley
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+corsixth.require("announcer")
+corsixth.require("entities.humanoids.staff")
+
+local AnnouncementPriority = _G["AnnouncementPriority"]
+
+--! A Doctor, Nurse, Receptionist, Handyman, or Surgeon
+class "Handyman" (Staff)
+
+---@type Handyman
+local Handyman = _G["Handyman"]
+
+--!param ... Arguments to base class constructor.
+function Handyman:Handyman(...)
+  self:Staff(...)
+end
+
+function Handyman:leaveAnnounce()
+  local announcement_priority = AnnouncementPriority.High
+  self.world.ui:playAnnouncement("sack006.wav", announcement_priority)
+end
+
+function Handyman:die()
+  if self.task then
```

Comment:
This self.task appears in 3 different places, could be refactored into another method.

Reply:
Will do